synthesizer/process/src/stack/helpers/synthesize.rs

```diff
@@ -100,6 +103,8 @@ impl<N: Network> Stack<N> {
         // Insert the proving key.
         self.insert_proving_key(function_name, proving_key)?;
         // Insert the verifying key.
-        self.insert_verifying_key(function_name, verifying_key)
+        self.insert_verifying_key(function_name, verifying_key)?;
```

Comment:
should we remove the proving key if the insertion of the verifying key fails (`insert_verifying_key` is fallible now)?

Reply:
`insert_verifying_key` was always fallible, we just don't return the Result yet because we have another function to check.

Indeed its not super clean that this doesn't happen atomically, but the only reason for failure at the moment is that the pk/vk/count already exists.

So I think we can leave it as is.